be/src/io/cache/block_file_cache.cpp

```diff
@@ -2097,6 +2160,367 @@ void BlockFileCache::update_ttl_atime(const UInt128Wrapper& hash) {
+    // consume loads of RAMs.
+    // Instead, we use protobuf serialize each of the single entry
+    // and provide the version field in the footer for upgrade
+    doris::io::cache::LruDumpEntryPB entry;
+    auto* hash_pb = entry.mutable_hash();
+    hash_pb->set_high(hash.high());
+    hash_pb->set_low(hash.low());
+    entry.set_offset(offset);
+    entry.set_size(size);
+
+    std::string serialized;
+    if (!entry.SerializeToString(&serialized)) {
+        std::string warn_msg = fmt::format("dump lru serialize failed, file={}, hash={}, offset={}",
+                                           filename, hash.to_string(), offset);
+        LOG(WARNING) << warn_msg;
+        return Status::InternalError<false>(warn_msg);
+    }
+
+    out.write(serialized.data(), serialized.size());
+#endif
+    return check_ofstream_status(out, filename);
+}
+
+Status BlockFileCache::finalize_dump(std::ofstream& out, size_t entry_num,
+                                     std::string& tmp_filename, std::string& final_filename,
+                                     size_t& file_size) {
+    // write footer: size_t entry_num, version, magic, totally 12 bytes
+    int8_t version = 1;
+    std::string magic_str = "DOR";
+    out.write(reinterpret_cast<const char*>(&entry_num), sizeof(entry_num));
+    out.write(reinterpret_cast<const char*>(&version), sizeof(version));
+    out.write(magic_str.c_str(), magic_str.size());
+    RETURN_IF_ERROR(check_ofstream_status(out, tmp_filename));
+    out.close();
+
+    // rename tmp to formal file
+    if (std::rename(tmp_filename.c_str(), final_filename.c_str()) != 0) {
+        std::remove(tmp_filename.c_str());
+        file_size = std::filesystem::file_size(final_filename);
+    }
+    return Status::OK();
+}
+
+void BlockFileCache::run_background_lru_dump() {
+    while (!_close) {
+        int64_t interval_ms = config::file_cache_background_lru_dump_interval_ms;
+        {
+            std::unique_lock close_lock(_close_mtx);
+            _close_cv.wait_for(close_lock, std::chrono::milliseconds(interval_ms));
+            if (_close) {
+                break;
+            }
+        }
+
+        auto dump_queue = [&](LRUQueue& queue, const std::string& queue_name) {
+            Status st;
+            std::vector<std::tuple<UInt128Wrapper, size_t, size_t>> elements;
+            elements.reserve(config::file_cache_background_lru_dump_tail_record_num);
+
+            {
+                std::lock_guard<std::mutex> lru_log_lock(_mutex_lru_log);
+                size_t count = 0;
+                for (const auto& [hash, offset, size] : queue) {
+                    if (count++ >= config::file_cache_background_lru_dump_tail_record_num) break;
+                    elements.emplace_back(hash, offset, size);
+                }
+            }
+
+            // Write to disk
+            int64_t duration_ns = 0;
+            std::uintmax_t file_size = 0;
+            {
+                SCOPED_RAW_TIMER(&duration_ns);
```

Comment:
no file header for the dumpped file?

Reply:
no, no header. but the footer will satisfy .